app/lib/state_change_notifier.rb

```diff
@@ -23,6 +26,33 @@ def self.submit_application(application_form)
     send(message, url)
   end
 
+  def self.accept_offer(accepted:, withdrawn: [], declined: [])
+    accepted_msg = "#{accepted.application_form.first_name} has accepted #{accepted.offered_option.course.provider.name}’s offer on #{accepted.offered_option.course.name_and_code}"
+
+    if withdrawn.any?
+      withdrawn_msg = "withdrawn their #{'application'.pluralize(withdrawn.count)} for "
+      withdrawals = withdrawn.map do |ac|
+        "#{ac.offered_option.course.name_and_code} at #{ac.offered_option.course.provider.name}"
+      end
+
+      withdrawn_msg += withdrawals.to_sentence
+    end
+
+    if declined.any?
+      declined_msg = 'declined '
+      declines = declined.map do |ac|
+        "#{ac.offered_option.course.provider.name}’s offer on #{ac.offered_option.course.name_and_code}"
```

Comment:
Is `offer on x` the right wording here? I'd expect it to read `offer for x`. (Same thing for accepted message above)

Reply:
Good point — looks like we use `for` elsewhere. I'll update :+1:

Reply:
Updated